File: pubnub-gson/pubnub-gson-docs/src/main/java/com/pubnub/docs/publishAndSubscribe/publish/PublishOld.java
package com.pubnub.docs.publishAndSubscribe.publish;

import com.google.gson.JsonArray;
import com.google.gson.JsonObject;
import com.pubnub.api.PubNubException;
import com.pubnub.api.java.PubNub;
import com.pubnub.api.models.consumer.PNPublishResult;
import com.pubnub.docs.SnippetBase;
import org.json.JSONObject;

import java.util.Arrays;

public class PublishOld extends SnippetBase {

    private void publishOldMessageToChannel() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#publish-a-message-to-a-channel-1

        PubNub pubNub = createPubNub();

        // snippet.publishOld
        JsonObject position = new JsonObject();
        position.addProperty("lat", 32L);
        position.addProperty("lng", 32L);

        System.out.println("before pub: " + position);
        pubNub.publish()
                .message(position)
                .channel("my_channel")
                .async(result -> {
                    result.onSuccess((PNPublishResult publishResult) -> {
                        System.out.println("pub timetoken: " + publishResult.getTimetoken());
                    });
                });
        // snippet.end
    }

    private void publishOldWithMetadata() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#publish-with-metadata-1

        PubNub pubNub = createPubNub();

        // snippet.publishOldWithMetadata
        PNPublishResult pnPublishResult = pubNub.publish()
                .message(Arrays.asList("hello", "there"))
                .channel("suchChannel")
                .shouldStore(true)
                .meta("meta") // optional meta data object which can be used with the filtering ability.
                .usePOST(true)
                .sync();
        // snippet.end
    }

    private void publishJsonObjectGson() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#publishing-jsonobject-google-gson-1

        PubNub pubNub = createPubNub();

        // snippet.publishJsonObjectGson
        JsonObject position = new JsonObject();
        position.addProperty("lat", 32L);
        position.addProperty("lng", 32L);

        System.out.println("before pub: " + position);
        pubNub.publish()
                .message(position)
                .channel("my_channel")
                .async(result -> {
                    result.onSuccess((PNPublishResult publishResult) -> {
                        System.out.println("pub timetoken: " + publishResult.getTimetoken());
                    });
                });
        // snippet.end
    }

    private void publishJsonArrayGson() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#publishing-jsonarray-google-gson-1

        PubNub pubNub = createPubNub();

        // snippet.publishJsonArrayGson
        JsonArray position = new JsonArray();
        position.add(32L);

        System.out.println("before pub: " + position);
        pubNub.publish()
                .message(position)
                .channel("my_channel")
                .async(result -> {
                    result.onSuccess((PNPublishResult publishResult) -> {
                        System.out.println("pub timetoken: " + publishResult.getTimetoken());
                    });
                });
        // snippet.end
    }

    private void publishJsonObjectOrgJson() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#publishing-jsonobject-orgjson-1

        PubNub pubNub = createPubNub();

        // snippet.publishJsonObjectOrgJson
        JSONObject position = new JSONObject();
        position.put("lat", 32L);
        position.put("lng", 32L);

        System.out.println("before pub: " + position);
        pubNub.publish()
                .message(position.toMap())
                .channel("my_channel")
                .async(result -> {
                    result.onSuccess((PNPublishResult publishResult) -> {
                        System.out.println("pub timetoken: " + publishResult.getTimetoken());
                    });
                });
        // snippet.end
    }

    private void publishJsonArrayOrgJson() throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#publishing-jsonarray-orgjson-1

        PubNub pubNub = createPubNub();

        // snippet.publishJsonArrayOrgJson
        org.json.JSONArray position = new org.json.JSONArray();
        position.put(32L);

        System.out.println("before pub: " + position);
        pubNub.publish()
                .message(position.toList())
                .channel("my_channel")
                .async(result -> {
                    result.onSuccess((PNPublishResult publishResult) -> {
                        System.out.println("pub timetoken: " + publishResult.getTimetoken());
                    });
                });
        // snippet.end
    }

    private void publishMessagesFor10Hours () throws PubNubException {
        // https://www.pubnub.com/docs/sdks/java/api-reference/publish-and-subscribe#store-the-published-message-for-10-hours-1

        PubNub pubNub = createPubNub();

        // snippet.publishMessagesFor10Hours
        PNPublishResult result = pubNub.publish()
                .channel("coolChannel")
                .message("test")
                .shouldStore(true)
                .ttl(10)
                .sync();
        // snippet.end
    }
}

File: pubnub-kotlin/pubnub-kotlin-docs/src/main/kotlin/com/pubnub/docs/publishAndSubscribe/publish/PublishOld.kt
package com.pubnub.docs.publishAndSubscribe.publish

import com.google.gson.JsonArray
import com.google.gson.JsonObject
import com.pubnub.docs.SnippetBase
import org.json.JSONArray
import org.json.JSONObject

class PublishOld : SnippetBase() {
    private fun publishMessageToChannel() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#basic-usage-10

        val pubnub = createPubNub()

        // snippet.publishMessageToChannel
        pubnub.publish(
            message = JsonObject().apply {
                addProperty("lat", 32L)
                addProperty("lng", 32L)
            },
            channel = "my_channel"
        ).async { result ->
            result.onSuccess { value ->
                println("Publish timetoken ${value.timetoken}")
            }
        }
        // snippet.end
    }

    private fun publishWithMetadata() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#publish-with-metadata-1

        val pubnub = createPubNub()

        // snippet.publishWithMetadata
        pubnub.publish(
            message = mapOf("hello" to "there"),
            channel = "my_channel",
            shouldStore = true,
            meta = mapOf("lang" to "en"),
            usePost = true
        ).async { result ->
            // check result
        }
        // snippet.end
    }

    private fun publishJsonObjectGson() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#publishing-jsonobject-google-gson

        val pubnub = createPubNub()

        // snippet.publishJsonObjectGson
        pubnub.publish(
            message = JsonObject().apply {
                addProperty("lat", 32L)
                addProperty("lng", 32L)
            },
            channel = "my_channel"
        ).async { result ->
            // check result
        }
        // snippet.end
    }

    private fun publishJsonArrayGson() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#publishing-jsonarray-google-gson-1

        val pubnub = createPubNub()

        // snippet.publishJsonArrayGson
        pubnub.publish(
            message = JsonArray().apply {
                add(32L)
                add(35L)
            },
            channel = "my_channel"
        ).async { result ->
            // check result
        }
        // snippet.end
    }

    private fun publishJsonObjectJsonOrg() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#publishing-jsonobject-orgjson-1

        val pubnub = createPubNub()

        // snippet.publishJsonObjectJsonOrg
        pubnub.publish(
            message = JSONObject().apply {
                put("lat", 32L)
                put("lng", 32L)
            },
            channel = "my_channel"
        ).async { result ->
            // check result
        }
        // snippet.end
    }

    private fun publishJsonArrayJsonOrg() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#publishing-jsonarray-orgjson-1

        val pubnub = createPubNub()

        // snippet.publishJsonArrayJsonOrg
        pubnub.publish(
            message = JSONArray().apply {
                put(32L)
                put(33L)
            },
            channel = "my_channel"
        ).async { result ->
            // check result
        }
        // snippet.end
    }

    private fun publishAndStoreMessagesFor10Hours() {
        // https://www.pubnub.com/docs/sdks/kotlin/api-reference/publish-and-subscribe#store-the-published-message-for-10-hours-1

        val pubnub = createPubNub()

        // snippet.publishAndStoreMessagesFor10Hours
        pubnub.publish(
            message = "test",
            channel = "my_channel",
            shouldStore = true,
            ttl = 10
        ).async { result -> // check result }
            // snippet.end
        }
    }
}
